torch/nn/parallel/parallel_apply.py

```diff
@@ -16,12 +20,18 @@ def parallel_apply(modules, inputs, kwargs_tup=None):
     lock = threading.Lock()
     results = {}
 
+    def _get_device(obj):
+        if isinstance(obj, Variable):
+            return torch.cuda.device_of(obj)
+        if isinstance(obj, collections.Iterable):
+            vals = obj.values() if isinstance(obj, collections.Mapping) else obj
+            return next(dev for dev in map(_get_device, vals) if dev is not None)
```

Comment:
If no inputs are given, it should mean that all devices are going to be used. I think it's a reasonable default.